hazelcast/src/test/java/com/hazelcast/util/executor/LoggingScheduledExecutorTest.java

```diff
@@ -79,20 +79,23 @@ public void no_remaining_task_after_cancel() throws Exception {
         executor = new LoggingScheduledExecutor(logger, 1, factory);
 
         for (int i = 0; i < 10; i++) {
-            Future<Integer> future = executor.submit(new Callable<Integer>() {
+            Future<Integer> future = executor.schedule(new Callable<Integer>() {
                 @Override
                 public Integer call() throws Exception {
                     TimeUnit.HOURS.sleep(1);
                     return null;
                 }
-            });
+            }, 1, HOURS);
 
             future.cancel(true);
         }
 
         BlockingQueue<Runnable> workQueue = ((LoggingScheduledExecutor) executor).getQueue();
 
-        assertEquals(0, workQueue.size());
+        // {@link LoggingScheduledExecutor.DEFAULT_PURGE_PERIOD}
+        sleepSeconds(10 + 2);
```

Comment:
`assertTrueEventually` can be used here also seems a 12 seconds length test can be marked with `@Category(SlowTest.class)`

Reply:
Will do :+1: